ui/components/multichain/asset-picker-amount/asset-picker-modal/AssetList.tsx

```diff
@@ -139,3 +128,56 @@ export default function AssetList({
     </Box>
   );
 }
+
+type TokenListItemWithBalanceProps = {
+  chain: any; // Replace 'any' with the actual type of 'chain' if available
+  token: AssetWithDisplayData<ERC20Asset> | AssetWithDisplayData<NativeAsset>;
+  primaryCurrency: string;
+  secondaryCurrency: string;
+  primaryNumberOfDecimals: number;
+  secondaryNumberOfDecimals: number;
+  nativeCurrency: string;
+  isDisabled: boolean;
+};
+
+function TokenListItemWithBalance({
+  chain,
+  token,
+  primaryCurrency,
+  secondaryCurrency,
+  primaryNumberOfDecimals,
+  secondaryNumberOfDecimals,
+  nativeCurrency,
+  isDisabled,
+}: TokenListItemWithBalanceProps) {
+  const { chainId } = chain.network;
+  const balancesByChainId = useSelector(
+    getSelectedAccountCachedBalancesByChainId,
+  );
+  const balanceValue = balancesByChainId?.[chainId];
+
+  console.log({ balancesByChainId });
+
+  const [, primaryCurrencyProperties] = useCurrencyDisplay(balanceValue, {
+    numberOfDecimals: primaryNumberOfDecimals,
+    currency: primaryCurrency,
+  });
+  const [secondaryCurrencyDisplay] = useCurrencyDisplay(balanceValue, {
+    numberOfDecimals: secondaryNumberOfDecimals,
+    currency: secondaryCurrency,
+    hideLabel: true,
+  });
+
+  return (
+    <TokenListItem
+      title={nativeCurrency}
+      chain={chain}
+      primary={primaryCurrencyProperties.value}
+      tokenSymbol={primaryCurrency}
+      secondary={secondaryCurrencyDisplay}
+      tokenImage={token.image}
+      isOriginalTokenSymbol
+      tooltipText={isDisabled ? 'swapTokenNotAvailable' : undefined}
+    />
+  );
+}
```

Comment:
Is there a reason why these changes can't be incorporated into the `TokenListItem` component? This seems to essentially wrap that component.